zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -205,18 +237,42 @@ impl<F: Field> MPTConfig<F> {
                     ifx! {not!(f!(q_not_first)) => {
                         require!(a!(is_start) => true);
                     }};
+                    cb.base.require_zero(
+                        "only one state is valid at a time",
+                        a!(is_start) * a!(is_branch) * a!(is_account) * a!(is_storage)
+                    );
```

Comment:
I think you mean `a!(is_start) + a!(is_branch) + a!(is_account) + a!(is_storage) => 1` because what you're checking here will be valid as long as only 1 is non-zero.

Also, the good news is that `matchx!` will kind of already do this check (as well as check that those values are boolean)! I say kind of because the `matchx!` does have a `_` so they are are also all allowed to be `0`.